### PR TITLE
Log failed transactions into beholder

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: core/capabilities/targets/write_target.go [hunk 1]
```diff
@@ -309,7 +319,40 @@ func (cap *WriteTarget) Execute(ctx context.Context, rawRequest capabilities.Cap
...
+	defer tick.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return capabilities.CapabilityResponse{}, nil
+		case <-tick.C:
+			txStatus, err := cap.cw.GetTransactionStatus(ctx, txID.String())
+			if err != nil {
+				cap.lggr.Errorw("Failed to get transaction status", "request", request, "transaction", txID, "err", err)
+				continue
```
Comment (vyzaldysanchez) [inline core/capabilities/targets/write_target.go:331]: I'm curious on whether we should really retry or not... What happens if it hangs retrying?

cc: @cedric-cordenier

File: core/capabilities/targets/write_target.go [hunk 2]
```diff
@@ -31,6 +37,9 @@ type WriteTarget struct {
 	receiverGasMinimum uint64
 	capabilities.CapabilityInfo
 
+	// emitter is used to emit messages from the WASM module to a configured collector.
```
Comment (cedric-cordenier) [inline core/capabilities/targets/write_target.go]: @vyzaldysanchez I think the comment is wrong here :)

File: core/capabilities/targets/write_target.go [hunk 3]
```diff
@@ -309,7 +319,40 @@ func (cap *WriteTarget) Execute(ctx context.Context, rawRequest capabilities.Cap
...
+					platform.KeyWorkflowID, request.Metadata.WorkflowID,
+					platform.KeyWorkflowName, request.Metadata.WorkflowName,
+					platform.KeyWorkflowOwner, request.Metadata.WorkflowOwner,
+					platform.KeyWorkflowExecutionID, request.Metadata.WorkflowExecutionID,
+				).Emit(ctx, msg)
+				if err != nil {
+					cap.lggr.Errorf("failed to send custom message with msg: %s, err: %v", msg, err)
+				}
+				return capabilities.CapabilityResponse{}, fmt.Errorf("submitted transaction failed: %w", err)
+			default:
```
Comment (cedric-cordenier) [inline core/capabilities/targets/write_target.go:350]: Can we log here: unexpected transaction status

File: core/capabilities/targets/write_target.go (hunk 1 above)
Comment (cedric-cordenier) [reply core/capabilities/targets/write_target.go:331]: That's fine: we'll retry every tick and eventually time out: the method should be context aware and should respect the timeout if it does block.

**Review by ettec — APPROVED** [review]

File: core/capabilities/targets/write_target.go (hunk 1 above)
Comment (patrickhuie19) [reply core/capabilities/targets/write_target.go:331]: @patrick-dowell I thought you worked on a similar PR to this a couple weeks ago - did we fail to find consensus on your approach?

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**